clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinatorTest.java

```diff
@@ -1053,6 +1058,49 @@ public void testForceMetadataRefreshForPatternSubscriptionDuringRebalance() {
         assertFalse(coordinator.rejoinNeededOrPending());
     }
 
+    @Test
+    public void testForceMetadataDeleteForPatternSubscriptionDuringRebalance() {
```

Comment:
Nice test to simulate the scenario you faced.

Reply:
In fact, only onJoinPrepare called the maybeAutoCommitOffsetsAsync method before.
Follow guozhang's suggestion , I will delete the maybeAutoCommitOffsetsSync(timer) method,
close() also calls maybeAutoCommitOffsetsAsync